product_harmonized_system_per_country/models/product_product.py

```diff
@@ -0,0 +1,39 @@
+#  Copyright (c) 2024 Groupe Voltaire
+#  @author Emilie SOUTIRAS  <emilie.soutiras@groupevoltaire.com>
+#  License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    hs_code = fields.Char(compute="_compute_hs_code", store=False)
+
+    @api.depends(
+        "hs_code_id", "hs_code_id.parent_id", "hs_code_id.child_ids", "categ_id"
+    )
+    def _compute_hs_code(self):
+        for pdt_tmpl in self:
+            if pdt_tmpl.hs_code_id:
+                pdt_tmpl.hs_code = (
+                    pdt_tmpl.hs_code_id.filter_per_country().hs_code or ""
+                )
+            else:
+                pdt_tmpl.hs_code = (
+                    pdt_tmpl.categ_id.get_hs_code_recursively().hs_code or ""
+                )
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def get_hs_code_recursively(self):
+        res = self.env["hs.code"]
+        if self:
+            self.ensure_one()
+            if self.hs_code_id:
+                res = self.hs_code_id.filter_per_country()
+            else:
+                res = super().get_hs_code_recursively()
+        return res
```

Comment:
It is almost the same method in product.product and product.category, is it possible to do a mixin and avoid duplicate method ?

Reply:
These functions are defined in the product_harmonized_system module, the mixin cannot be done here.